src/cpr_data_access/models.py

```diff
@@ -686,3 +693,71 @@ def get_all_text_blocks(
                         output_values.append(block)
 
         return output_values
+
+    def _create_text_block_doc(self, doc: AnyDocument, block: TextBlock) -> dict[str, Any]:
```

Comment:
This should live outside of the `Dataset` class and I think a name like `_create_text_block_and_document_dict` might be clearer